test/v3/controllers/SubmitFinalDeclarationControllerSpec.scala

```diff
@@ -78,10 +84,12 @@ class SubmitFinalDeclarationControllerSpec
 
   }
 
-  private val rawData     = SubmitFinalDeclarationRawData(nino, taxYear, calculationId)
-  private val requestData = SubmitFinalDeclarationRequest(Nino(nino), TaxYear.fromMtd(taxYear), calculationId)
+  private val rawData                     = SubmitFinalDeclarationRawData(nino, taxYear, calculationId)
+  private val requestData                 = SubmitFinalDeclarationRequest(Nino(nino), TaxYear.fromMtd(taxYear), CalculationId(calculationId))
+  private val retrieveDetailsRequestData  = RetrieveCalculationRequest(Nino(nino), TaxYear.fromMtd(taxYear), CalculationId(calculationId))
+  private val retrieveDetailsResponseData = minimalCalculationResponse
 
-  "submit final declaration" should {
+  "V3-submit final declaration" should {
```

Comment:
We don't need to specify v3 as this is a v3 spec. We should also use the name of the controller for example, 
```suggestion
  "SubmitFinalDeclarationController" should {
```